src/main/java/org/zalando/problem/spring/web/advice/security/ProblemAuthenticationEntryPoint.java

```diff
@@ -14,8 +15,13 @@
 @Component
 public class ProblemAuthenticationEntryPoint implements AuthenticationEntryPoint {
 
+    private final HandlerExceptionResolver resolver;
+
     @Autowired
-    private HandlerExceptionResolver resolver;
+    public ProblemAuthenticationEntryPoint(
+            @Qualifier("handlerExceptionResolver") final HandlerExceptionResolver resolver) {
```

Comment:
In this case a comment (e..g on the class) might be helpful for future generations.

> AuthenticationEntryPoint forwarding authentication exceptions to spring-webmvc's HandlerExceptionResolver as defined by WebMvcConfigurationSupport. Compatible with spring-webmvc 4.3.3.

This might help someone in case the Spring guys decide to rename the bean.